### PR TITLE
Use validator moniker to show block proposer

### LINKED ISSUE #223: Use validator moniker to show block proposer.
Add `getProposerMoniker` function in `BlockHook.Block.t` that takes a block and a list of `ValidatorHook.Validator.t` and return the moniker string (or "Unknown" if unknown). 

Use this function in all blocks page and other pages that require showing moniker.

### PR DESCRIPTION
**Pull request by prin-r**
Fixes #223

### CODE REVIEW

**Review by sorawit — CHANGES_REQUESTED** [review]
Please also fix report detail. Check with @evilpeach

**Review by sorawit — APPROVED** [review]
lgtm

File: scan/src/components/ReportTable.re [hunk 1]
```diff
@@ -34,6 +40,14 @@ let make = (~reports: list(RequestHook.Report.t)) => {
     </THead>
     {reports
      ->Belt.List.map(({reporter, txHash, reportedAtHeight, reportedAtTime, values}) => {
+         let moniker =
+           validators
+           ->Belt_List.keepMap(validator =>
+               validator.operatorAddress |> Address.toHex == (reporter |> Address.toHex)
```
Comment (evilpeach) [inline scan/src/components/ReportTable.re]: try `==` or make `compare` function of Address

File: scan/src/rpc/BlockHook.re [hunk 2]
```diff
@@ -20,6 +20,17 @@ module Block = {
     };
 
   let decodeBlocks = json => JsonUtils.Decode.(json |> list(decodeBlock));
+
+  let getProposerMoniker = (block: t, validators: list(ValidatorHook.Validator.t)) =>
+    validators
+    ->Belt_List.keepMap(validator =>
+        validator.consensusPubkey
+        |> PubKey.toAddress
+        |> Address.toHex == (block.proposer |> Address.toHex)
```
Comment (evilpeach) [inline scan/src/rpc/BlockHook.re]: make `cmp` function

**Review by evilpeach — APPROVED** [review]
LGTM

**Pull request merged**